cluster/test/setup.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+set -aeuo pipefail
+
+echo "Running setup.sh"
+echo "Creating cloud credential secret..."
+${KUBECTL} -n upbound-system create secret generic provider-secret --from-literal=credentials="${UPTEST_CLOUD_CREDENTIALS}" --dry-run=client -o yaml | ${KUBECTL} apply -f -
```

Comment:
Is there a reason we dry-run and then pipe to an apply instead of running without the `--dry-run` option?

Reply:
yes, if I don't include that it fails when secret exists.

```
kubectl create secret generic aws-secret -n upbound-system --from-file=creds=/Users/hasanturken/.aws/credentials -o yaml | kubectl apply -f -
error: failed to create secret secrets "aws-secret" already exists
error: no objects passed to apply
```